### ISSUE TITLE
Snooze diagnostics

### ISSUE THREAD

**Issue by JohanLarsson** [dotnet/roslyn#39683]
![image](https://user-images.githubusercontent.com/1640096/68163342-9c8aea80-ff5a-11e9-8e2b-95010843a9aa.png)

In the picture above `IDE0009` is not yet fluent in c# 8. This happens with most analyzers when language features are added. The problem is that disabling the rule for this reason means one must remember to enable it again which means one will forget.

The snooze feature would suppress the diagnostic in `.ruleset` or in `#pragma` for the current version of the analyzer.

**Comment by jnm2** [comment]
I'm expecting to replace .ruleset with the [new editorconfig capabilities](https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-preview#net-productivity-164P1) in 16.4:

![](https://docs.microsoft.com/en-us/visualstudio/releases/2019/media/16.4/ruleseverityineditor.png)

What about `dotnet_diagnostic.IDE0009.suppress_until = 2020-01-01` or something like that?

**Comment by ryzngard** [comment]
Copying some thought process from discord: 

Date may be complicated, because it doesn't require user action to start causing problems. Version of the analyzer as mentioned above seems like a pragmatic solution from bouncing thoughts around. 

cc @mavasani

**Comment by mavasani** [comment]
NOTE: If implemented, this would have to be implemented in the compiler layer, as the configuration seems to be applicable even for analyzer execution in CI.

**Comment by CyrusNajmabadi** [comment]
agreed that just using our existing mechanisms for disabling things is the way to go here.
